### PR TITLE
[TT-5506] fix OAS servers in patch

### PR DESCRIPTION
**Pull request by jeffy-mathew**
[changelog]
internal: fix OAS servers in patch endpoint

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4083/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2396699663)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4083/merge
Commit: 90d6b5b2a61aea25bc50e8b207fc9fa5185a35ed 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2396751372)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4083/merge
Commit: 9c6308783d9b10ef55ea83884a9198a2b127db55 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2396903830)

**Review by titpetric — APPROVED** [review]

File: apidef/oas/default.go [hunk 1]
```diff
@@ -223,3 +224,17 @@ func getQueryValPtr(val string) *bool {
...
+
+// RetainOldServerURL retains the first entry from old servers provided
+// tyk adds a server URL to the start of oas.Servers to add the gw URL
+// RetainOldServerURL can be used when API def is patched.
+func RetainOldServerURL(oldServers, newServers openapi3.Servers) openapi3.Servers {
+	if len(oldServers) > 0 && len(newServers) > 0 {
+		if oldServers[0].URL == newServers[0].URL {
+			return newServers
+		}
+		newServers = append(openapi3.Servers{oldServers[0]}, newServers...)
```
[the remark this answers is not in the document]
Comment (jeffy-mathew) [reply apidef/oas/default.go:236]: The uniqueness will be checked in `updateServers` func and that too for the ones which Tyk adds, if there are any other duplicates, it's something we should consider the user decided to keep it as it is. right?

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4083/merge
Commit: 6e7738d84dc72b0b6a098d701d4a13df429ad1f6 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2407093787)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4083/merge
Commit: e3420ca26ceb169e2cb01c9b12aa57187d11d2c4 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2407152298)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 4b5f5c401324283766bc8b15bdfd60aac43a2ee6 [TT-5506] fix OAS servers in patch (#4083)

* fix OAS servers in patch

[changelog]
internal: fix OAS servers in patch endpoint
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2407239546)

**Pull request merged**